gammapy/maps/sparse.py

```diff
@@ -0,0 +1,169 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import copy
+import numpy as np
+from ._sparse import find_in_array, merge_sparse_arrays
+
+
+def slices_to_idxs(slices, shape, ndim):
+
+    if slices == Ellipsis:
+        slices = tuple([slice(None)] * ndim)
+    elif not isinstance(slices, tuple):
+        slices = tuple([slices])
+
+    #nslice = min(1, sum([not isinstance(s, slice) for s in slices]))
+    #nslice += sum([isinstance(s, slice) for s in slices])
+    nslice = len(slices)
+    idim = None
+    idx = []
+    for i, s in enumerate(slices):
+
+        si = [None] * nslice
+        if isinstance(s, slice):
+            si[i] = slice(None)
+            start = 0 if s.start is None else s.start
+            stop = shape[i] if s.stop is None else s.stop
+
+            idx += [np.arange(start, stop, 1)[si]]
+        else:
+            if idim is None:
+                idim = i
+            si[idim] = slice(None)
+            idx += [np.array(s, ndmin=1)[si]]
+
+    return idx
+
+
+class SparseArray(object):
+    """Sparse N-dimensional array object.  This class implements a data
```

Comment:
Is `SparseArray` a class end-users will interact with, or is it supposed to be something internal that doesn't need to be fully documented?

If it's something end-users see, please put an empty line after the first summary line of the docstring.
Also, please add a `Parameters` section documenting what the parameters of `__init__` are.
Maybe even throw in an `Examples` section with a few lines of example code showing how to make one of those and do something with it?

Reply:
In principle users shouldn't need to interact with `SparseArray` directly -- the recommended interface for manipulating maps is via get/set/fill methods of the map classes.  However it's probably good to have it well documented in any case since developers will want to know how to use this class when working with the maps classes.  It's also possible that there may be use cases for it outside of `gammapy.maps`.